huaweicloud/services/dms/data_source_huaweicloud_dms_rocketmq_broker.go

```diff
@@ -0,0 +1,128 @@
+package dms
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chnsz/golangsdk"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func DataSourceDmsRocketMQBroker() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: resourceDmsRocketMQBrokerRead,
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"instance_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"brokers": {
+				Type:        schema.TypeList,
+				Elem:        DmsRocketMQBrokerBrokerSchema(),
+				Computed:    true,
+				Description: `Indicates the list of the brokers.`,
+			},
+		},
+	}
+}
+
+func DmsRocketMQBrokerBrokerSchema() *schema.Resource {
+	sc := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"ids": {
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeInt},
+				Computed:    true,
+				Description: `Indicates the all broker ids.`,
+			},
+			"broker_name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the name of a broker.`,
+			},
+		},
+	}
+	return &sc
+}
+
+func resourceDmsRocketMQBrokerRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+
+	var mErr *multierror.Error
```

Comment:
remove this line to line:105

Reply:
the code is automatically generated, the style is unified